scalding-core/src/test/scala/com/twitter/scalding/ExecutionTest.scala

```diff
@@ -262,6 +263,35 @@ class ExecutionTest extends WordSpec with Matchers {
       assert(oldCounters != newCounters, "With new configs given the source changed we shouldn't cache so the counters should be different")
 
     }
+
+    "correctly add cached file into config" in {
+      val execution = Execution.withCached("/path/to/your/file.txt") { cachedFile =>
+        Execution.getConfig.map { config =>
+          config.getDistributedCachedFiles should contain only cachedFile
+        }
+      }
+
+      execution.waitFor(Config.default, Hdfs(strict = true, new Configuration(false))) match {
+        case Success(s) => s
+        case Failure(e) => fail(s"Failed running execution, exception:\n$e")
+      }
+    }
+
+    "correctly add cached files into config" in {
+      val execution =
+        Execution.withCached("/path/to/your/one.txt") { one =>
+          Execution.withCached("/path/to/your/second.txt") { second =>
+            Execution.getConfig.map { config =>
+              config.getDistributedCachedFiles should contain only (one, second)
+            }
+          }
+        }
+
+      execution.waitFor(Config.default, Hdfs(strict = true, new Configuration(false))) match {
```

Comment:
@johnynek what do you mean "get the file"? tests above test `getDistributedCachedFiles` and `withCached`, they should actually execute `Execution.withCached`.